### DISCUSSION TITLE
Skyrim SE works with Mod Organizer 2 and SKSE

### DISCUSSION THREAD

**Discussion by thmrtnz** [The-Wineskin-Project/WineskinServer#251] [General]
I am able to run Skyrim SE with Mod Organizer 2 and SKSE without visual glitches.
With these wrappers and engines : Wineskin-2.9.1.1 + WS11WineCX64Bit21.1.0
I have set through winetricks d3dcompiler, d3dx11, vcrun2019, dxvk1101, faudio and dotnet48 and steam.
I have used SKSE for AE build 2.1.5 (game version 1.6.353). SkyUI works. Things appear to work properly so far.

Only hiccup is the mouse sensitivity and behavior. It is very floaty at the moment and stops in its movement at times.

![Screen Shot 2022-04-22 at 20 31 30](https://user-images.githubusercontent.com/47714003/164776857-1dc3a0a3-4143-4f8f-b5eb-7f43b9233b5b.png)

**Comment by Gcenx** [comment]
The mouse acting up is related to mouse capture, in fullscreen the game would require accessibility permissions for it to have permission to grab the mouse.

I believe CX21.2 back-ported the upstream change for mouse capture to using an Apple private API that doesn’t require any special permissions for mouse capture, I’ve not yet had the time to really test out the new Engine.

**Comment by thmrtnz** [comment]
Wabbajack runs too. Edit: and crashes!

![Screen Shot 2022-04-22 at 23 41 02](https://user-images.githubusercontent.com/47714003/164799699-f89d9953-db56-425c-866c-1d07beaf6962.png)

**Reply by Gcenx** [reply]
You might need to force that exe to use WineD3D instead of DXVK.

**Reply by thmrtnz** [reply]
Unfortunately without DXVK it simply goes full black screen. I have to hard reboot.
I tried a wrapper with DXVK and Skyrim and a wrapper without DXVK and Wabbajack
and a relative symlink to share my steam library.
Will keep digging, I might resort to a VM for that.

**Reply by Gcenx** [reply]
Your saying after forcing Wabbajack.exe to use WineD3D it’s a black screen?

You can that exe specifically to not sure DXVK and leave everything else using DXVK. You need to add the exe into winecfg and after selecting it you’ll be able to add overrides for that exe only, this includes setting a different windows versions of needed.

**Reply by thmrtnz** [reply]
Using DXVK it starts and I get a window and it might crash hashing the game files or eventually downloading the modlists. Without DXVK, it goes to black screen.

Its debug log reads:
`[0424/122623.535:ERROR:network_change_notifier_win.cc(228)] WSALookupServiceBegin failed with: 0
[0424/122624.413:ERROR:dxva_video_decode_accelerator_win.cc(1440)] DXVAVDA fatal error: could not LoadLibrary: msmpeg2vdec.dll: Module not found. (0x7E)
[0424/122624.634:ERROR:dxva_video_decode_accelerator_win.cc(1406)] DXVAVDA fatal error: Could not load msmpeg2vdec.dll: Module not found. (0x7E)
[0424/122625.497:ERROR:network_change_notifier_win.cc(228)] WSALookupServiceBegin failed with: 0`

I have dotnet 4.8 and 5, since it uses CefSharp, I might be missing informations.

[debug.log](https://github.com/Gcenx/WineskinServer/files/8549504/debug.log)
[LastRunWine.log](https://github.com/Gcenx/WineskinServer/files/8549506/LastRunWine.log)
[Wabbajack 2022-04-24 12_26_30.log](https://github.com/Gcenx/WineskinServer/files/8549515/Wabbajack.2022-04-24.12_26_30.log)
[Wabbajack 2022-04-24 12_45_51.log](https://github.com/Gcenx/WineskinServer/files/8549516/Wabbajack.2022-04-24.12_45_51.log)
[Wabbajack.current.log](https://github.com/Gcenx/WineskinServer/files/8549517/Wabbajack.current.log)

**Reply by thmrtnz** [reply]
No DXVK: 
`
0430:err:winediag:wined3d_init_vkd3d_once Failed to load libvkd3d-shader.
0430:err:eventlog:ReportEventW L"Application: CefSharp.BrowserSubprocess.exe\nFramework Version: v4.0.30319\nDescription: The process was terminated due to an unhandled exception.\nException Info: System.AccessViolationException
`

[LastRunWine.log](https://github.com/Gcenx/WineskinServer/files/8549620/LastRunWine.log)
[debug.log](https://github.com/Gcenx/WineskinServer/files/8549636/debug.log)

**Reply by Gcenx** [reply]
msmpeg2vdec is part of mfplat no wonder one of the those workaround might install the missing needed dlls, use the “Winehq engine launch” and use one of the two mfplat installers

**Reply by thmrtnz** [reply]
It's working so far, though it keeps asking for msvproc
![Screen Shot 2022-04-24 at 17 36 45](https://user-images.githubusercontent.com/47714003/164984275-acf7fa28-16af-4292-8d3b-6093610f2d0a.png)

I used [z0z0z](https://github.com/z0z0z)[mf-install](https://github.com/z0z0z/mf-install) editing the script with calls to grealpath instead of realpath and gcp instead of cp, coreutils must be installed with homebrew

**Reply by thmrtnz** [reply]
Seems it works

**Reply by thmrtnz** [reply]
I believe the Battle.net launcher works with the very same stuff, cefsharp, d3d11, msmpeg2vdec and msvproc ;
lots of chatter on reddit and co about people stuck there

**Reply by Gcenx** [reply]
Maybe I should expand https://github.com/Gcenx/winetricks/commit/5b21b7f9d7f7ca9c6420221be3e522a4e28b6969 to install those additinal dlls then we should need to use those installers

**Reply by thmrtnz** [reply]
It struggle no more with msvproc if you pause the gallery autoplay, less subject to crash
![Screen Shot 2022-04-24 at 18 27 22](https://user-images.githubusercontent.com/47714003/164986613-bf6057eb-6c86-4924-b138-95c0bb1bd362.png)

**Reply by thmrtnz** [reply]
![Screen Shot 2022-04-24 at 18 41 59](https://user-images.githubusercontent.com/47714003/164986950-6cba062d-3ce1-49d2-94ee-54ebb526b1f6.png)

**Reply by Gcenx** [reply]
I’ve not seen mention of msvproc being part of media foundation before.

**Reply by Gcenx** [reply]
> It's working so far, though it keeps asking for msvproc ![Screen Shot 2022-04-24 at 17 36 45](https://user-images.githubusercontent.com/47714003/164984275-acf7fa28-16af-4292-8d3b-6093610f2d0a.png)
> 
> I used [z0z0z](https://github.com/z0z0z)[mf-install](https://github.com/z0z0z/mf-install) editing the script with calls to grealpath instead of realpath and gcp instead of cp, coreutils must be installed with homebrew

Just an FWI there’s no longer an need to use mf-install the mf verb now does the same job as mf-install and the cab version see https://github.com/The-Wineskin-Project/winetricks/pull/1

**Comment by thmrtnz** [comment]
I've added a wiki page for Skyrim on Apple Silicon with Wineskin.

**Reply by Gcenx** [reply]
I’m thinking of just dedicating a repository to a wiki that may make is easier.

**Reply by Gcenx** [reply]
Just an FYI, Steam _must_ be installed via winetricks or there will be missing required overrides for known bugs.

**Reply by thmrtnz** [reply]
Yes. 
```
    # Not all users need this disabled, but let's play it safe for now
    if w_workaround_wine_bug 22053 "Disabling gameoverlayrenderer to prevent game crashes on some machines."; then
        w_override_dlls disabled gameoverlayrenderer
    fi

    if w_workaround_wine_bug 44985 "Disabling libglesv2 to make Store and Library function correctly."; then
        w_override_dlls disabled libglesv2
    fi

    # Otherwise Steam Store and Library don't show
    w_call corefonts


```
You can use Winetricks, set up steam in program files, or move it before launching it and it installs, or install it first where you want and run winetricks to apply the overrides. Or am I doing it terribly wrong?

**Reply by Gcenx** [reply]
It’s best to stick to installing using winetricks the less steps an end user needs to make the better.

You never want to move after the instal as the install location is saved to the registry.

**Comment by thmrtnz** [comment]
I could play yesterday and now there's a "EXCEPTION_ACCESS_VIOLATION" for d3d11.dll. Uh.  
Well. I've rebuilt the whole thing twice, can play a couple times and then get EXCEPTION_ACCESS_VIOLATION. 
Edit> I've changed settings in the inis, using BethINI. Something **something is breaking on poor preset.**  

it must be the modlist I'm running, or Mod Organizer. The game launches fine on its own via Steam  

[crash-2022-08-07-06-03-39.log](https://github.com/Gcenx/WineskinServer/files/9276281/crash-2022-08-07-06-03-39.log)
[crash-2022-08-07-06-08-30.log](https://github.com/Gcenx/WineskinServer/files/9276282/crash-2022-08-07-06-08-30.log)
[SkyrimSE_d3d11.log](https://github.com/Gcenx/WineskinServer/files/9276322/SkyrimSE_d3d11.log)
[SkyrimSE_dxgi.log](https://github.com/Gcenx/WineskinServer/files/9276323/SkyrimSE_dxgi.log)

**Reply by thmrtnz** [reply]
Something in the settings of the game causes this.
I will test it all.

**Reply by thmrtnz** [reply]
With Bethini medium settings it runs.
At **poor preset** it crashes on start.

**Reply by Gcenx** [reply]
That’s really weird

**Reply by thmrtnz** [reply]
I believe it happens when I toggle off shadows.

**Comment by PitNox** [comment]
(I have no idea if this is the right thread) I am trying to follow this guide but I'm not super proficient in wineskin and I'm struggling to understand how to fix my issues.
- I have managed to install and run Steam successfully
- I have downloaded and installed Skyrim Special Edition
- Clicking "Play" on Steam opens the launcher and detects my system settings (Ultra High)
- I click Play in the Launcher but the game insta-crashes (icon pops-up un dock for a split second then disappears)

**I have no idea how to check logs for this** as I am running Steam as a "Custom exe" from inside the "Skyrim" wrapper and I have no idea how to interact with this custom exe as "show package contents" on it shows nothing but the barebones wineskin resources and the "CustomEXE" app.
Any help is very much appreciated!

I consider myself pretty tech savvy and I have extensive experience with Skyrim inner workings due to ten years or so of modding (most of which on a mac before the "32bit massacre" of Catalina), but this is just a step prior so I have no idea how to move or what to try :(

**Comment by NateSaxo** [comment]
Issue: wine process for MO2 would start but no window or application would open. Installing MO2 version 2.4.1 seems to have fixed this specific problem, for now.
